funsor/adjoint.py

```diff
@@ -79,11 +72,7 @@ def adjoint(self, red_op, bin_op, root, targets):
             for v, adjv in in_adjs.items():
                 adjoint_values[v] = bin_op(adjoint_values[v], adjv)
 
-        target_adjs = {}
-        for v in targets:
-            target_adjs[v] = adjoint_values[v]
-
-        return target_adjs
+        return {target: adjoint_values[target] for target in targets}
```

Comment:
For `target` not in `adjoint_values` this returns 1 (`ops.UNITS[bin_op]`) but it should be 0 or undefined I think.